src/julia.h

```diff
@@ -1328,6 +1325,12 @@ typedef enum {
     JL_IMAGE_JULIA_HOME = 1,
     //JL_IMAGE_LIBJULIA = 2,
 } JL_IMAGE_SEARCH;
+#ifdef JULIA_ENABLE_THREADING
+// this helps turn threading compilation mismatches into linker errors
+#define julia_init julia_init__threading
+#define jl_init jl_init__threading
```

Comment:
So ... we un-exported jl_init by default without even a NEWS entry?

![image](https://user-images.githubusercontent.com/327706/27045703-c44570e2-4f6f-11e7-89c3-dedcbd87662f.png)

Reply:
This is rather annoying when you dynamically load libjulia.so and don't know whether to call jl_init or jl_init__threading to initialize Julia.